fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java

```diff
@@ -187,4 +193,35 @@ public ScalarOperator visitInPredicate(InPredicateOperator predicate, ScalarOper
 
         return isIn ? Utils.compoundOr(result) : Utils.compoundAnd(result);
     }
+
+    // rewrite collection element to subfiled
+    @Override
+    public ScalarOperator visitCollectionElement(CollectionElementOperator collectionElement,
+                                                 ScalarOperatorRewriteContext context) {
+        if (collectionElement.getChild(0).getType().isStructType()) {
+            Preconditions.checkState(collectionElement.getChild(1).isConstantRef());
+            Preconditions.checkState(collectionElement.getChild(1).getType().isIntegerType());
```

Comment:
Is these check necessary?

Reply:
emmm, I think unnecessary, just avoid person write error code

Reply:
Maybe add some error msg is better? If not, the unknown error return to client is really confused.